kotlin/display-grid/src/main/java/com/esri/arcgisruntime/sample/displaygrid/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+/* Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.esri.arcgisruntime.sample.displaygrid
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.SpatialReference
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.Grid
+import com.esri.arcgisruntime.mapping.view.LatitudeLongitudeGrid
+import com.esri.arcgisruntime.mapping.view.MgrsGrid
+import com.esri.arcgisruntime.mapping.view.UsngGrid
+import com.esri.arcgisruntime.mapping.view.UtmGrid
+import com.esri.arcgisruntime.symbology.LineSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.TextSymbol
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.popup_menu.view.*
+
+class MainActivity : AppCompatActivity() {
+  private var lineColor = 0
+  private var labelColor = 0
+  private var labelPosition = Grid.LabelPosition.ALL_SIDES
+
+  private val center: Point by lazy {
+    Point(
+      -7702852.905619,
+      6217972.345771,
+      SpatialReference.create(3857)
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // create a map with imagery basemap
+      map = ArcGISMap(Basemap.createImagery()).also { map ->
+        map.initialViewpoint = Viewpoint(center, 23227.0)
+      }
+      // set defaults on grid
+      mapView.grid = LatitudeLongitudeGrid()
+    }
+
+    // set up a popup menu to manage grid settings
+    val builder =
+      AlertDialog.Builder(this@MainActivity)
+    val popupView = layoutInflater.inflate(R.layout.popup_menu, null)
```

Comment:
No need to inflate this. Just give popup_menu an id (see the xml file comment) and adjust the import statement above (see above).

Reply:
This isn't working for me...are you sure it doesn't need to be inflated? Without inflating it, it seems that the popupView just stays null

Reply:
you are right, I conflated inflation with findViewById().  sorry about that 😓

Reply:
no worries! is it all good to go then?